openpype/hosts/maya/plugins/publish/extract_look.py

```diff
@@ -46,59 +56,302 @@ def find_paths_by_hash(texture_hash):
     return legacy_io.distinct(key, {"type": "version"})
 
 
-def maketx(source, destination, args, logger):
-    """Make `.tx` using `maketx` with some default settings.
+@six.add_metaclass(ABCMeta)
+class TextureProcessor:
+    def __init__(self, log=None):
+        if log is None:
+            log = logging.getLogger(self.__class__.__name__)
+        self.log = log
 
-    The settings are based on default as used in Arnold's
-    txManager in the scene.
-    This function requires the `maketx` executable to be
-    on the `PATH`.
+    def apply_settings(self, system_settings, project_settings):
+        pass
 
-    Args:
-        source (str): Path to source file.
-        destination (str): Writing destination path.
-        args (list): Additional arguments for `maketx`.
-        logger (logging.Logger): Logger to log messages to.
+    @abstractmethod
+    def process(self,
+                source,
+                colorspace,
+                color_management,
+                staging_dir):
+        """Process the `source` texture.
+
+        Must be implemented on inherited class.
+
+        This must always return a TextureResult even when it does not generate
+        a texture. If it doesn't generate a texture then it should return a
+        TextureResult using the input path and colorspace.
+
+        Args:
+            source (str): Path to source file.
+            colorspace (str): Colorspace of the source file.
+            color_management (dict): Maya Color management data from
+                `lib.get_color_management_preferences`
+            staging_dir (str): Output directory to write to.
+
+        Returns:
+            TextureResult: The resulting texture information.
+
+        """
+        pass
+
+    @staticmethod
+    def get_extension():
+        pass
+
+    def __repr__(self):
+        # Log instance as class name
+        return self.__class__.__name__
+
+
+class MakeRSTexBin(TextureProcessor):
+    """Make `.rstexbin` using `redshiftTextureProcessor`"""
+
+    def process(self,
+                source,
+                colorspace,
+                color_management,
+                staging_dir):
+        """
+        with some default settings.
+
+        This function requires the `REDSHIFT_COREDATAPATH`
+        to be in `PATH`.
+
+        Args:
+            source (str): Path to source file.
+
+        """
+        if "REDSHIFT_COREDATAPATH" not in os.environ:
+            raise RuntimeError("Must have Redshift available.")
```

Comment:
it seems that this code is not working to check if REDSHIFT_COREDATAPATH exists in the environment. 
![image](https://user-images.githubusercontent.com/64118225/227837682-ff9be75d-60df-4d71-90af-18d6ce6163ae.png)
But it works when I run with the separate scripts, when checking the environment. 
![image](https://user-images.githubusercontent.com/64118225/227838067-4222aa41-71d5-426e-80bb-88ee7fc8abc1.png)

Is it a good idea to include this in the validator instead of the extract to validate whether there is the reshift setup in the user varaible.

Reply:
You might need to restart OpenPype tray or your machine completely before OP picks up that variable.

Also, haven't tested the rstex code yet and I expect it to error. However, it should pick up the tool fine I believe.

Could you check the actual error message?

Reply:
Fixed with: https://github.com/ynput/OpenPype/pull/4711/commits/35f76a2365d76aa97ca32a5591e831662dcd4029